File: laythe_vm/fixture/py_benchmark/zoo.py
import time

class Zoo:
  def __init__(self):
    self.aarletk  = 1
    self.baboon   = 1
    self.cat      = 1
    self.donkey   = 1
    self.elephant = 1
    self.fox      = 1

  def ant(self):
    return self.aarletk

  def banana(self):
    return self.baboon

  def tuna(self):
    return self.cat

  def hay(self):
    return self.donkey

  def grass(self):
    return self.elephant

  def mouse(self):
    return self.fox

zoo = Zoo()
sum = 0
start = time.time()

while sum < 10000000:
  sum += (zoo.ant() \
      + zoo.banana() \
      + zoo.tuna() \
      + zoo.hay() \
      + zoo.grass() \
      + zoo.mouse())

print(sum)
print(time.time() - start)

File: laythe_vm/fixture/rb_benchmark/zoo.rb
class Zoo
  def initialize
    @aarletk  = 1
    @baboon   = 1
    @cat      = 1
    @donkey   = 1
    @elephant = 1
    @fox      = 1
  end

  def ant
    @aarletk
  end

  def banana
    @baboon
  end

  def tuna
    @cat
  end

  def hay
    @donkey
  end

  def grass
    @elephant
  end

  def mouse
    @fox
  end
end

zoo = Zoo.new
sum = 0
start = Time.new

while sum < 10000000 do
  sum += zoo.ant() + zoo.banana() + zoo.tuna() + zoo.hay() + zoo.grass() + zoo.mouse()
end

puts(sum)
puts(Time.new - start)
